pkgs/applications/science/electronics/signalhound-spike/default.nix

```diff
@@ -0,0 +1,149 @@
+{ lib
+, stdenv
+, fetchurl
+, fetchFromGitHub
+, unzip
+, autoPatchelfHook
+, fontconfig
+, freetype
+, udev
+, libICE
+, libSM
+, libX11
+, libXcursor
+, libXext
+, libXi
+, libXfixes
+, libXrandr
+, libXrender
+, libxcb
+, xcbutilrenderutil
+, xcbutilimage
+, xcbutilkeysyms
+, xcbutilwm
+, libxkbcommon
+, libGL
+, libusb
+, libpulseaudio
+, libpng
+, harfbuzz
+, zlib
+, icu60
+, double-conversion
+, glib
+, makeWrapper
+}:
+
+let
+double-conversion1 = double-conversion.overrideAttrs (old: rec {
+  version = "2.0.4";
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "double-conversion";
+    rev = "${version}";
+    hash = "sha256-EHgFvHNF16GphJO1tWLir26cLe1srSo4vZULd1oHX/I=";
+  };
+});
+in stdenv.mkDerivation rec {
+  pname = "signalhound-spike";
+  version = "3.9.0";
+
+  src = fetchurl {
+    url = "https://signalhound.com/sigdownloads/Spike/Spike(Ubuntu22.04x64)_${(lib.replaceStrings ["."] ["_"] version)}.zip";
+    hash = "sha256-mEZbRU+81odHJ4X1a1t7LRSWcpfkPGCCBqiVI4ZtEXM=";
+  };
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+  buildInputs = [ double-conversion1 ];
+
+  rpath = lib.makeLibraryPath [
+    fontconfig
+    freetype
+    libICE
+    libSM
+    udev
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXrandr
+    libXrender
+    libXi
+    libxcb
+    xcbutilrenderutil
+    xcbutilimage
+    xcbutilkeysyms
+    xcbutilwm
+    libxkbcommon
+    libGL
+    libusb
+    libpulseaudio
+    libpng
+    harfbuzz
+    zlib
+    icu60
+    double-conversion1
+    glib
+  ]
+  + ":${stdenv.cc.cc.lib}/lib64";
+
+  installPhase = ''
```

Comment:
Make sure to run the preInstall and postInstall hooks with `runHook preInstall` and same for the post install